pkg/node/manager/manager.go

```diff
@@ -368,7 +368,11 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		// to create symmetric traffic when sending nodeIP->pod and pod->nodeIP.
 		if address.Type == addressing.NodeCiliumInternalIP || m.conf.EncryptionEnabled() ||
 			option.Config.EnableHostFirewall || option.Config.JoinCluster {
-			tunnelIP = nodeIP
+			if tunnelIP = n.GetNodeTunnelEndpointIPv4(); tunnelIP == nil {
+				log.Debugf("No tunnelIP manually defined, just use the nodeIP")
```

Comment:
nit: `Debugf()` uses the format variation, which implies that there are parameters coming. However, this is only a string. This could be equally done with `log.Debug(...)`.

Reply:
Removed the log function here.